pint_server/app.py

```diff
@@ -425,11 +439,35 @@ def get_provider_servers(provider):
             for server in servers]
 
 
+def trim_images_payload(images):
+    payload_size = jsonify(images=images).content_length
```

Comment:
Note that this operation represents wasted work, since we call `jsonify()` just to get the content length here... And this will be wasted work for every `/v1/<provider>/images` request, whether the request was for JSON or XML...

If this is a concern we probably should restructure things such that a trim function (or set of functions) is passed as an argument to the `make_response()` routine, and it does the `jsonify()`, and checks the content length of that, and only if that exceeds the MAX_PAYLOAD_SIZE do we call the trim function to do any trimming...

That would allow for making a trimming mechanism more generic, and encoding specific, e.g. we wouldn't trim XML unless the XML encoded output exceeded the payload size limit.

Reply:
Calling jsonify() give us an more accurate look at the actual payload size. It will be applicable to both XML and JSON payloads, as the XML payload is just a straight conversion from JSON.

Reply:
Per discussion elsewhere, it is desirable that the XML and JSON results be equivalent, so we will always want to determine whether the JSON results (since they seem to grow faster than XML for equivalent content) wouldn't exceed the payload limit... So it would only be wasted work for JSON queries where we will `jsonify()` the images list twice, discarding this one after having retrieved the content_length...

Reply:
I can't imagine jsonify() add a noticeable overhead though.

Reply:
Probably not in the case of relatively small data hierarchy, but walking the data structure and rendering it as a JSON string representation for a large data object does represent a body of work, given that in Python strings are immutable objects. The `jsonify()` is going to be a fairly optimised implementation, so the overhead will be minimised, but always something to keep in mind.

It is less of a concern since we have to do it anyway since we want the XML and JSON results to be equivalent, so we need to always `jsonfiy()` to get the content_length, so I don't really see a problem... I was just raising it initially as at first glance we were `jsonify()`ing the image list even for XML output requests, which seemed counter-intuitive until I realised that it may be desirable to ensure that the XML and JSON results would be equivalent.

As such I'm happy to resolve this discussion with no changes required.